components/bitcoin-topics/Navigation.tsx

```diff
@@ -77,32 +133,95 @@ export function Navigation({
-                        const childPost = posts.find(
-                            (p) => p.slug === childSlug
-                        )
-                        return childPost
-                            ? {
-                                  title: childPost.title,
-                                  href: `/${childPost.path}`,
-                                  project: childPost.project || false
-                              }
-                            : null
-                    })
-                    .filter(Boolean) as NavigationLink[]
-            }))
+    const navigation = useMemo(() => {
+        const navigation: NavigationCategory[] = sortedCategories.map(
+            (category) => ({
+                title: category,
+                links: groupedPosts[category].map((post) => ({
+                    title: post.title,
+                    href: `/${post.path}`,
+                    project: post.project || false,
+                    children: post.childTopics?.childSlugs
+                        ?.map((childSlug: string) => {
+                            // Explicitly typing `childSlug` as `string`
+                            const childPost = posts.find(
+                                (p) => p.slug === childSlug
+                            )
+                            return childPost
+                                ? {
+                                      title: childPost.title,
+                                      href: `/${childPost.path}`,
+                                      project: childPost.project || false
+                                  }
+                                : null
+                        })
+                        .filter(Boolean) as NavigationLink[],
+                    icon: post.icon
+                }))
+            })
+        )
+        return navigation
+    }, [posts, groupedPosts, sortedCategories])
+
+    const toggleTopic = useCallback((topicHref: string) => {
+        setExpandedTopics((prev) => ({
+            ...prev,
+            [topicHref]: !prev[topicHref]
+        }))
+    }, [])
+
+    // Update expanded state when pathname changes
+    useEffect(() => {
+        const newExpandedState: Record<string, boolean> = {}
+        navigation.forEach((section) => {
+            section.links.forEach((link) => {
+                if (
+                    link.children &&
+                    (link.href === pathname ||
+                        link.children.some((child) => child.href === pathname))
+                ) {
+                    newExpandedState[link.href] = true
+                }
+            })
+        })
+        setExpandedTopics((prev) => {
+            const updated = { ...prev, ...newExpandedState }
+            return JSON.stringify(updated) !== JSON.stringify(prev)
+                ? updated
+                : prev
         })
+    }, [pathname, navigation])
+
+    const getIcon = useCallback((iconName: string) => {
+        const IconComponent =
```

Comment:
Interesting piece of code here, is this IconComponent, global ?

Reply:
this allows dynamic access to any icon from react-icons. It's not global, it's scoped within the getIcon function

Reply:
Ohhh alright, since react icons can be imported anywhere I have no issues with it